src/views/virtualmachines/actions/components/VirtualMachineMigration/hooks/useMigratableStorageClasses.ts

```diff
@@ -0,0 +1,26 @@
+import { V1beta1StorageSpecAccessModesEnum } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { StorageProfileModelGroupVersionKind } from '@kubevirt-utils/models';
+import { getName } from '@kubevirt-utils/resources/shared';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+const useMigratableStorageClasses = (): string[] => {
+  const [storageProfiles] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: StorageProfileModelGroupVersionKind,
+    isList: true,
+  });
+
+  const allowedStorageProfilesNames = storageProfiles?.reduce((acc, storageProfile: any) => {
```

Comment:
In the #2524 I've assumed that volume/access mode from storage profile are only recommendations (user is allowed to overwrite). Perhaps we should inform the user that he can unblock the storage class by changing configuration in the storage profile or add a way to by-pass  storage profile?

Reply:
Currently we don't let the user customize anything during the storage migration so it will use the default settings.